Telerik.Sitefinity.Amazon/BlobStorage/AmazonBlobStorageProvider.cs

```diff
@@ -26,25 +26,42 @@ public class AmazonBlobStorageProvider : CloudBlobStorageProvider
         /// <param name="config">The collection of parameters (each by its name and value) of the current provider's configuration settings.</param>
         protected override void InitializeStorage(NameValueCollection config)
         {
+            var useIamInstanceRoleValue = "false";
+            if (config.AllKeys.Contains(UseIamInstanceRoleKey))
+                useIamInstanceRoleValue = config[UseIamInstanceRoleKey].Trim();
+
+            if (!bool.TryParse(useIamInstanceRoleValue, out this.useIamInstanceRole))
+                throw new ConfigurationException("'{0}' unable to parse {1}".Arrange(UseIamInstanceRoleKey, useIamInstanceRoleValue));
+
             this.accessKeyId = config[AccessKeyIdKey].Trim();
-            if (String.IsNullOrEmpty(this.accessKeyId))
+            if (!this.useIamInstanceRole && String.IsNullOrEmpty(this.accessKeyId))
                 throw new ConfigurationException("'{0}' is required.".Arrange(AccessKeyIdKey));
 
             this.secretKey = config[SecretKeyKey].Trim();
-            if (String.IsNullOrEmpty(this.secretKey))
+            if (!this.useIamInstanceRole && String.IsNullOrEmpty(this.secretKey))
                 throw new ConfigurationException("'{0}' is required.".Arrange(SecretKeyKey));
 
             this.bucketName = config[BucketNameKey].Trim();
             if (String.IsNullOrEmpty(this.bucketName))
                 throw new ConfigurationException("'{0}' is required.".Arrange(BucketNameKey));
 
+            if (config.AllKeys.Contains(KeyPrefixKey))
+                this.keyPrefix = config[KeyPrefixKey].Trim();
+
+            if (string.IsNullOrWhiteSpace(this.keyPrefix))
+                this.keyPrefix = "";
+
             string regionEndpointString = config[RegionEndpointKey].Trim();
             var endpointField = typeof(RegionEndpoint).GetField(regionEndpointString, BindingFlags.Static | BindingFlags.Public);
             if ((string.IsNullOrWhiteSpace(regionEndpointString)) || (endpointField == null))
                 throw new ConfigurationException("'{0}' is required.".Arrange(RegionEndpointKey));
 
             var regionEndpoint = (RegionEndpoint)endpointField.GetValue(null);
-            this.transferUtility = new TransferUtility(accessKeyId, secretKey, regionEndpoint);
+
```

Comment:
If we're using the EC2 Instance Profile Role, don't pass accesskey and secretaccesskey into the TransferUtility constructor, the AWS SDK will use the instance's role by default

Reply:
@BenWolstencroft, 
We've managed to test EC2InstanceProfile IAM functionality and as you said this works fine if Sitefinity is deployed to an EC2 instance, and the EC2 instance's profile role has the correct S3 Bucket policy applied. 
But what if useIamInstanceRoleValue is set to true and Sitefinity is hosted outside AWS (or EC2 instance's profile role in not attached to EC2 instance), then error is thrown:

Exception 1 of 4:
System.ArgumentException: App.config does not contain credentials information. Either add the AWSAccessKey and AWSSecretKey or AWSProfileName.
   at Amazon.Runtime.StoredProfileAWSCredentials..ctor(String profileName, String profilesLocation) in d:\Jenkins\jobs\build-sdk-v2\workspace\sdk\src\AWSSDK_DotNet35\Amazon.Runtime\AWSCredentials.cs:line 318
   at Amazon.Runtime.EnvironmentAWSCredentials..ctor() in d:\Jenkins\jobs\build-sdk-v2\workspace\sdk\src\AWSSDK_DotNet35\Amazon.Runtime\AWSCredentials.cs:line 590
   at Amazon.Runtime.FallbackCredentialsFactory.<Reset>b__1() in d:\Jenkins\jobs\build-sdk-v2\workspace\sdk\src\AWSSDK_DotNet35\Amazon.Runtime\AWSCredentials.cs:line 1117
   at Amazon.Runtime.FallbackCredentialsFactory.GetCredentials(Boolean fallbackToAnonymous) in d:\Jenkins\jobs\build-sdk-v2\workspace\sdk\src\AWSSDK_DotNet35\Amazon.Runtime\AWSCredentials.cs:line 1137


We need proper documentation, how exactly this setup works, and probably cover this scenario described above.